File: src/_as/01/clamp_resource.ts
// SSoT: src/ontology/math/clamp_resource.md
import { RESOURCE_MAX } from "../00/mod";

@inline
export function clamp_resource(value: i64): i32 {
if (value < 0) return 0;
if (value > (RESOURCE_MAX as i64)) return RESOURCE_MAX;
return value as i32;
}

File: src/00/sigma_core/src/ontology_gen/01/clamp_resource.rs
// SSoT: file:///Users/s0fractal/OMEGA/src/ontology/math/clamp_resource.md
#![allow(unused_imports)]
use super::super::L00::*;

pub fn clamp_resource(value: i64) -> i32 {
    if value < 0 {
        0
    } else if value > (RESOURCE_MAX as i64) {
        RESOURCE_MAX as i32
    } else {
        value as i32
    }
}
